internal/eea/eea_test.go

```diff
@@ -169,6 +171,194 @@ func createNeededEntities(ctx context.Context, t *testing.T) (projID uuid.UUID,
 	return proj.ID, repo.ID
 }
 
+func TestFlushAll(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name        string
+		mockDBSetup func(context.Context, *mockdb.MockStore)
+	}{
+		{
+			name: "flushes one repo",
```

Comment:
I don't want to block this PR, but it seems like we should have some more tests about the negative scenarios. For example, if I take a look at the code for `FlushAll` I think we should cover scenarios such as:

1) Asserting that calling the method when the cache is empty returns no error.
2) Returns an error when `GetRepositoryByID` fails
3) Does nothing when `GetRepositoryByID` cannot find a project for the repository.
4) Does nothing when `buildEntityWrapper` cannot find any matching data in the DB

Reply:
@dmjb was planning to submit that in a separate PR as to not overwhelm the reader. But yeah, I agree we need negative tests as well.